src/fabtrimino.c

```diff
@@ -0,0 +1,250 @@
+#include "fabtrimino.h"
+
+struct square
+{
+    enum shape shape;
+};
+
+struct fabtrimino
+{
+    enum shape shape;
+    struct square*** matrix;
+    struct vector* pos;
+};
+
+
+void nullify_squares_matrix(struct square*** mat, size_t size)
```

Comment:
Given what I mentioned earlier you'll probably be wondering how you declare your matrix as a parameter. The solution is pretty simple:
```
void nullify_squares_matrix(struct square mat[4][4]) {...}
nullify_squares_matrix(trimino->matrix);
```
I'm about 95% sure that this does the right thing but there's a slight chance it might copy the entire matrix in which case you might want to use this instead:
```
void nullify_squares_matrix(struct square *mat[4][4]) {...}
nullify_squares_matrix(&trimino->matrix);
```
I think this is right but mostly this is why I never use god damn multi-dimensional arrays. They get really weird.

Also since all you're doing is zeroing memory here and that we've flatten your entire matrix layout, then you can just use this to zero everything:
```
memset(trimino->matrix, 0, 4*4*sizeof(trimino->matrix[0][0]);
```
way simpler and faster as gcc uses vectorized ops to do the memset.